File: clients/python/examples/tables-create-vector-index-specify.py
from astrapy import DataAPIClient
from astrapy.constants import VectorMetric
from astrapy.info import TableVectorIndexOptions

# Get an existing table
client = DataAPIClient()
database = client.get_database(
    "**API_ENDPOINT**", token="**APPLICATION_TOKEN**"
)
table = database.get_table("**TABLE_NAME**")

# Index a vector column
table.create_vector_index(
    "**INDEX_NAME**",
    column="**VECTOR_COLUMN_NAME**",
    options=TableVectorIndexOptions(
        metric=VectorMetric.DOT_PRODUCT,
        source_model="nv-qa-4",
    ),
)

File: clients/typescript/examples/tables-create-vector-index-specify.ts
import { DataAPIClient } from "@datastax/astra-db-ts";

// Get an existing table
const client = new DataAPIClient();
const database = client.db("**API_ENDPOINT**", {
  token: "**APPLICATION_TOKEN**",
});
const table = database.table("**TABLE_NAME**");

// Index a vector column
(async function () {
  await table.createVectorIndex("**INDEX_NAME**", "**VECTOR_COLUMN_NAME**", {
    options: {
      metric: "dot_product",
      sourceModel: "nv-qa-4",
    },
  });
})();
